File: model/src/main/java/net/artux/pda/model/items/ItemsHelper.java
package net.artux.pda.model.items;

import java.io.Serializable;
import java.util.List;
import java.util.Optional;

public final class ItemsHelper implements Serializable {

    public static void add(List<ItemModel> list, ItemModel model) {
        if (model instanceof WearableModel)
            ((WearableModel) model).setEquipped(false);
        if (model.getType().isCountable()) {
            Optional<ItemModel> itemModel = list.stream()
                    .filter(item -> item.baseId == model.baseId)
                    .findFirst();
            if (itemModel.isPresent()) {
                ItemModel item = itemModel.get();
                item.setQuantity(item.getQuantity() + model.getQuantity());
            } else {
                list.add(model);
            }
        } else {
            model.setQuantity(1);
            list.add(model);
        }
    }

}

File: model/src/main/java/net/artux/pda/model/items/ItemsHelper.kt
package net.artux.pda.model.items

import java.io.Serializable

object ItemsHelper : Serializable {
    fun add(list: MutableList<ItemModel>, model: ItemModel) {
        if (model is WearableModel) model.isEquipped = false
        if (model.type.isCountable) {
            val itemModel = list.stream()
                .filter { item: ItemModel -> item.baseId == model.baseId }
                .findFirst()
            if (itemModel.isPresent) {
                val item = itemModel.get()
                item.quantity = item.quantity + model.quantity
            } else {
                list.add(model)
            }
        } else {
            model.quantity = 1
            list.add(model)
        }
    }
}